sdk/dapp-kit/src/hooks/wallet/useSignAndExecuteTransactionBlock.ts

```diff
@@ -74,42 +75,83 @@ export function useSignAndExecuteTransactionBlock({
 				);
 			}
 
-			if (executeFromWallet) {
-				const walletFeature = currentWallet.features['sui:signAndExecuteTransactionBlock'];
+			const canExecuteFromWallet =
+				!!currentWallet.features['sui:signAndExecuteTransactionBlock:v2'];
+
+			const reportEffects =
+				currentWallet.features['sui:reportTransactionBlockEffects']
+					?.reportTransactionBlockEffects ?? (() => {});
+
+			if (canExecuteFromWallet) {
+				const walletFeature = currentWallet.features['sui:signAndExecuteTransactionBlock:v2'];
 				if (!walletFeature) {
 					throw new WalletFeatureNotSupportedError(
 						"This wallet doesn't support the `signAndExecuteTransactionBlock` feature.",
 					);
 				}
 
-				return walletFeature.signAndExecuteTransactionBlock({
+				const result = await walletFeature.signAndExecuteTransactionBlock({
 					...signTransactionBlockArgs,
+					transactionBlock: {
+						async toJSON() {
```

Comment:
How are we feeling about this now that it's here?

Reply:
I don't love it, but I don't know that any of the other options are better?